tests/e2e/validatecluster/validatecluster_test.go

```diff
@@ -189,14 +184,51 @@ var _ = Describe("Verify Create", Ordered, func() {
 			Eventually(func(g Gomega) {
 				pods, _ := e2e.ParsePods(kubeConfigFile, false)
 				count := e2e.CountOfStringInSlice("test-daemonset", pods)
-				fmt.Println("POD COUNT")
-				fmt.Println(count)
-				fmt.Println("NODE COUNT")
-				fmt.Println(len(nodes))
 				g.Expect(len(nodes)).Should((Equal(count)), "Daemonset pod count does not match node count")
 			}, "420s", "10s").Should(Succeed())
 		})
 
+		It("Verifies Restart K3s", func() {
+			_, err := e2e.DeployWorkload("daemonset.yaml", kubeConfigFile, *hardened)
+			Expect(err).NotTo(HaveOccurred(), "Daemonset manifest not deployed")
+			nodes, _ := e2e.ParseNodes(kubeConfigFile, false)
+			Eventually(func(g Gomega) {
+				pods, _ := e2e.ParsePods(kubeConfigFile, false)
+				count := e2e.CountOfStringInSlice("test-daemonset", pods)
+				g.Expect(len(nodes)).Should((Equal(count)), "Daemonset pod count does not match node count")
+				podsRunning := 0
+				for _, pod := range pods {
+					if strings.Contains(pod.Name, "test-daemonset") && pod.Status == "Running" {
+						podsRunning++
+					}
```

Comment:
One more thing to add in here that I missed the first time around: `Ready`. Should be x/x (1/1 or 2/2 or whatever, just not something like 1/2 or 0/1). We should check **both here before the restart and after the restart** that all of the pods in this daemonset are also Ready.